cvat/apps/engine/models.py

```diff
@@ -219,7 +220,61 @@ class FloatArrayField(AbstractArrayField):
 class IntArrayField(AbstractArrayField):
     converter = int
 
+class ValidationMode(str, Enum):
+    GT = "gt"
+    GT_POOL = "gt_pool"
+
+    @classmethod
+    def choices(cls):
+        return tuple((x.value, x.name) for x in cls)
+
+    def __str__(self):
+        return self.value
+
+class ValidationParams(models.Model):
+    task_data = models.OneToOneField(
+        'Data', on_delete=models.CASCADE, related_name="validation_params"
+    )
+
+    mode = models.CharField(max_length=32, choices=ValidationMode.choices())
+
+    frame_selection_method = models.CharField(
+        max_length=32, choices=JobFrameSelectionMethod.choices()
+    )
+    random_seed = models.IntegerField(null=True)
+
+    frames: models.manager.RelatedManager[ValidationFrame]
+    frame_count = models.IntegerField(null=True)
+    frame_share = models.FloatField(null=True)
+    frames_per_job_count = models.IntegerField(null=True)
+    frames_per_job_share = models.FloatField(null=True)
+
+class ValidationFrame(models.Model):
+    validation_params = models.ForeignKey(
+        ValidationParams, on_delete=models.CASCADE, related_name="frames"
+    )
+    path = models.CharField(max_length=1024, default='')
+
+class ValidationLayout(models.Model):
```

Comment:
Layout:

1. [Merriam-Webster](https://www.merriam-webster.com/dictionary/layout) 
> the plan or design or arrangement of something laid out
2. [Wiki](https://en.wikipedia.org/wiki/Layout) 
> In general terms, a layout is a structured arrangement of items within certain limits, or a plan for such arrangement. Specifically, layout may refer to: Page layout, the arrangement of visual elements on a page.

It's used to describe validation frames in tasks, both for simple GT and for Honeypots. That's why it doesn't have `pool` in the name.

> What is the reason to bind ValidationParams and ValidationLayout to Data instead of Task model?

This is done to be the same as storing `deleted_frames` in the `Data` model. Basically, it describes task data.

Reply:
Layout sounds like some set of elements and their relation to each other. 
But in our case is just couple of sets. Pool would sound good to describe and it is applicable in general for both GT job and Honeypot job. 

Hovewer if you do not want to use the word **Pool** -> it is okay, up to you.

Reply:
> This is made to be the same as storing deleted_frames in the Data model.

I can't really understand the explanation. Hovewer in the future this design may be a problem if we want to use the same **Data** object to create multiple tasks (this is not a fact that we will do this, but anyway).

Reply:
Never mind, considering existing database layout feature like: **import raw data, select them and create tasks based on them** already not implementable without new database classes.